server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java

```diff
@@ -104,21 +107,47 @@ public GCRun(Ample.DataLevel level, ServerContext context) {
   }
 
   @Override
-  public Iterator<String> getCandidates() {
+  public Iterator<GcCandidate> getCandidates() {
     return context.getAmple().getGcCandidates(level);
   }
 
+  /**
+   * Removes gcCandidates from the metadata location depending on type.
+   *
+   * @param gcCandidates Collection of deletion reference candidates to remove.
+   * @param type type of deletion reference candidates.
+   */
+  @Override
+  public void deleteGcCandidates(Collection<GcCandidate> gcCandidates, GcCandidateType type) {
+    if (inSafeMode()) {
+      System.out.println("SAFEMODE: There are " + gcCandidates.size()
+          + " reference file gcCandidates entries marked for deletion from " + level + ".\n"
+          + "          Examine the log files to identify them.\n");
+      log.info("SAFEMODE: Listing all ref file gcCandidates for deletion");
+      for (GcCandidate gcCandidate : gcCandidates) {
+        log.info("SAFEMODE: {}", gcCandidate);
+      }
+      log.info("SAFEMODE: End reference candidates for deletion");
+    }
+
+    if (!config.getBoolean(Property.GC_REMOVE_IN_USE_CANDIDATES)
+        && type.equals(GcCandidateType.INUSE)) {
+      return;
+    }
```

Comment:
Feel free to ignore this comment because I am not sure if its workable, makes sense, or maybe its already done.  I was wondering if it would make sense to move this check out of GCRun and into GarbageCollectionAlgorithm somehow.  If possible it may make this check more easily testable by the unit test for the GarbageCollectionAlgorithm.  Maybe the check is still here for validation purposes, not sure.

Reply:
I had to add a new method to GarbageCollectionEnvironment interface but I was able to bring this property check up to the GCA. 

It is now unit tested via the SimpleGarbageCollectorTest